pkg/webhook/policy.go

```diff
@@ -290,24 +293,19 @@ func (h *validationHandler) getValidationMessages(res []*rtypes.Result, req *adm
 				eventMsg = "Admission webhook \"validation.gatekeeper.sh\" denied request"
 				reason = "FailedAdmission"
 			}
-			ref := getViolationRef(
-				h.gkNamespace,
-				req.AdmissionRequest.Kind.Kind,
-				resourceName,
-				req.AdmissionRequest.Namespace,
-				r.Constraint.GetKind(),
-				r.Constraint.GetName(),
-				r.Constraint.GetNamespace())
-			h.eventRecorder.AnnotatedEventf(
-				ref,
-				annotations,
-				corev1.EventTypeWarning,
-				reason,
-				"%s, Resource Namespace: %s, Constraint: %s, Message: %s",
-				eventMsg,
-				req.AdmissionRequest.Namespace,
-				r.Constraint.GetName(),
-				r.Msg)
+
+			enamespace := h.gkNamespace
+			if *admissionEventsInvolvedNamespace && len(req.AdmissionRequest.Namespace) > 0 {
+				enamespace = req.AdmissionRequest.Namespace
+			}
+
+			ref := getViolationRef(enamespace, req.AdmissionRequest.Kind.Kind, resourceName, obj.GetResourceVersion(), obj.GetUID())
+
+			if *admissionEventsInvolvedNamespace || len(req.AdmissionRequest.Namespace) == 0 {
```

Comment:
Same here, should `admissionEventsInvolvedNamespace` be false here?

Reply:
This just changes the way the event message is output slightly. It checks if the events are generated in the involved objects namespace (flag true) or this is a cluster scoped resource (no namespace) then we do not need to output the resources namespace in the events message... imo this is because

1. When the flag is enabled and we view an event, the event is already generated in the resources namespace so it's redundant. 
2. Cluster scoped events don't have a namespace